### PR TITLE
Upgrade to shopify dependencies

### PR DESCRIPTION
**Pull request by lemonmade**
This PR is a first step towards removing the `@remote-ui/web-workers` package as described in https://github.com/Shopify/remote-ui/issues/96. My plan is to move this library to Quilt, which uses a newer version of the `@sewing-kit` packages (now called [loom](https://github.com/Shopify/loom)) to manage its development tools. To make that transition easier, this PR moves the remote-ui repo to use all those same Shopify tools and configurations. This results in a small number of code changes and a huge number of linter/ config file changes, but hopefully better that it happens here rather than as I am moving the packages around. This also kind of finishes off the transfer of this repo from my personal GitHub account, where I originally built it alongside `@sewing-kit`.

### CODE REVIEW

File: loom.config.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {createWorkspace} from '@shopify/loom';
+import {eslint} from '@shopify/loom-plugin-eslint';
+import {prettier} from '@shopify/loom-plugin-prettier';
+import {workspaceTypeScript} from '@shopify/loom-plugin-typescript';
+import {jest} from '@shopify/loom-plugin-jest';
+
+export default createWorkspace((workspace) => {
+  workspace.use(
+    workspaceTypeScript(),
+    jest(),
```
[the remark this answers is not in the document]
Comment (lemonmade) [reply loom.config.ts:10]: Right, I think I saw this but I didn't understand it. What does setting up Jest have to do with building a library? I know building TypeScript is kind of implicitly a build step too but it's mostly not build-related in my head too.

**Review by vividviolet — COMMENTED** [review]
Can we exclude test files - currently the type files are being generated:
image

**Comment by BPScott** [comment]
> Can we exclude test files

I would have thought you'd want your test files to be type-checked and to report any errors?

You can make sure they're not published by excluding them in your package.json's files array. See [polaris for an example](https://github.com/Shopify/polaris-react/blob/main/package.json#L16-L20)

File: loom.config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lemonmade) [reply loom.config.ts:10]: But we expect everyone to lint, right? Now I just have three dependencies I need to maintain and manually configure myself. That doesn't seem better form me as a consumer.

**Pull request merged**